rebasebot/bot.py

```diff
@@ -83,8 +82,9 @@ def _commit_go_mod_updates(gitwd: git.Repo, source: GitHubBranch) -> None:
             )
             logging.debug("go mod tidy output: %s", proc.stdout.decode())
 
-            # Only run go mod vendor if a vendor folder already exists
-            if os.path.exists(os.path.join(module_base_path, "vendor")):
+            # Only run go mod vendor if a vendor folder already exists, or we are in the project root.
+            # The way in which we configure go build in Dockerfiles expects to find a `/vendor` on the project root.
+            if os.path.exists(os.path.join(module_base_path, "vendor")) or module_base_path == ".":
```

Comment:
i'm wondering what will happen with something like the autoscaler repo where we don't have a top level vendor folder. this is especially relevant since upstream is going to remove the vendor folder from the cluster-autoscaler subdir. 

fwiw, the autoscaler repo does not have a top level vendor folder and i'm not sure if the go tooling will succeed there.

Reply:
We discussed this offline, and now want to run `go mod vendor` for all the `go.mod` we find.
Updated:: https://github.com/openshift-eng/rebasebot/pull/43/commits/0bad75f7da1f9e03f9d0d2d3591d4cc085ff8992